lib/libc/common/source/time/asctime_r.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <zephyr/sys/util.h>
+
+#define DATE_STRING_BUF_SZ 26
+
+char *asctime_r(const struct tm *tp, char *buf)
+{
+	static const char wday_str[7][3] = {
+		"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat",
+	};
+	static const char mon_str[12][3] = {
+		"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
+	};
+
+	if ((tp == NULL) || (buf == NULL) || (tp->tm_wday >= ARRAY_SIZE(wday_str)) ||
```

Comment:
tm_wday and tm_mon are both signed int; I think you should use an explicit cast here to make the comparison against the array size clearly correct.

glibc uses "???" for the weekday and month values when the incoming data are out of bounds. This ensures that the resulting buffer is always filled with something and not left uninitialized. glibc isn't completely consistent here; year values which are wildly out of range will leave the buffer unfilled.

Reply:
added an explicit cast to the `int`s, and simply return `NULL` when it's OOB.

The caller should refer to the return value, is it necessary for the function to initialize the buffer? If so I can probably add an additional `buf[0] = '\0'`